src/Volume.elm

```diff
@@ -0,0 +1,171 @@
+US liquid gallons, imperial fluid ounces etc. It is stored as a number of cubic meters.
+
+@docs Volume, CubicMeters
+
+
+## Metric
+
+@docs cubicMeters, inCubicMeters
+@docs milliliters, inMilliliters, liters, inLiters 
+
+
+## Imperial
+
+@docs cubicInches, inCubicInches, cubicFeet, inCubicFeet, cubicYards, inCubicYards
+@docs usLiquidGallons, inUsLiquidGallons, usDryGallons, inUsDryGallons, imperialGallons, inImperialGallons
+--@docs usLiquidQuarts, usDryQuarts, imperialQuarts
+--@docs usLiquidPints, usDryPints, imperialPints
+--@docs usFluidOunces, imperialFluidOunces
+
+-}
+
+import Length exposing (Meters)
+import Quantity exposing (Quantity(..), Cubed)
+
+
+{-| -}
+type alias CubicMeters =
+    Cubed Meters
+
+
+{-| -}
+type alias Volume =
+    Quantity Float CubicMeters
+
+
+{-| Construct a volume from a number of cubic meters.
+-}
+cubicMeters : Float -> Volume
+cubicMeters numCubicMeters =
+    Quantity numCubicMeters
+
+
+{-| Convert a volume to a number of cubic meters.
+-}
+inCubicMeters : Volume -> Float
+inCubicMeters (Quantity numCubicMeters) =
+    numCubicMeters
+
+
+{-| Construct a volume from a number of cubic inches.
+-}
+cubicInches : Float -> Volume
+cubicInches numCubicInches =
+    cubicMeters (0.0254 * 0.0254 * 0.0254 * numCubicInches)
+
+
+{-| Convert a volume to a number of cubic inches.
+-}
+inCubicInches : Volume -> Float
+inCubicInches volume =
+    inCubicMeters volume / (0.0254 * 0.0254 * 0.0254)
+
+
+{-| Construct a volume from a number of cubic feet.
+-}
+cubicFeet : Float -> Volume
+cubicFeet numCubicFeet =
+    cubicMeters (0.3048 * 0.3048 * 0.3048 * numCubicFeet)
+
+
+{-| Convert a volume to a number of cubic feet.
+-}
+inCubicFeet : Volume -> Float
+inCubicFeet volume =
+    inCubicMeters volume / (0.3048 * 0.3048 * 0.3048)
+
+
+{-| Construct a volume from a number of cubic yards.
+-}
+cubicYards : Float -> Volume
+cubicYards numCubicYards =
+    cubicMeters (0.9144 * 0.9144 * 0.9144 * numCubicYards)
+
+
+{-| Convert a volume to a number of cubic yards.
+-}
+inCubicYards : Volume -> Float
+inCubicYards volume =
+    inCubicMeters volume / (0.9144 * 0.9144 * 0.9144)
+
+
+{-| Construct a volume from a number of milliliters.
+-}
+milliliters : Float -> Volume
+milliliters numMilliliters =
+    cubicMeters (1.0e-6 * numMilliliters)
+
+
+{-| Convert a volume to a number of milliliters.
+-}
+inMilliliters : Volume -> Float
+inMilliliters volume =
+    1.0e6 * inCubicMeters volume
+
+
+{-| Construct a volume from a number of liters.
+-}
+liters : Float -> Volume
+liters numLiters =
+    cubicMeters (0.001 * numLiters)
+
+
+{-| Convert a volume to a number of liters.
+-}
+inLiters : Volume -> Float
+inLiters volume =
+    1000 * inCubicMeters volume
+
+
+{-| Construct a volume from a number of usLiquidGallons.
+-}
+usLiquidGallons : Float -> Volume
+usLiquidGallons numUsLiquidGallons =
+    cubicMeters ( numUsLiquidGallons / 264.17220000000003 )
```

Comment:
This might be a bit too extreme - but no harm done. @ianmackenzie Are you using any particular definitive doc to get the conversion factors?

Reply:
Typically I've just been scouring Wikipedia - for U.S. liquid gallons is there a reason not to use `numUsLiquidGallons *  0.003785411784`? https://en.wikipedia.org/wiki/Gallon#US_liquid_gallon

I'm open to using a single definitive source if there's a good one...

Reply:
Yeah, I guess it would good to find one reference - I'll try to evaluate some options. If you have any candidates, let me know!

Reply:
Have had a little look around - nothing jumps out. Countries and disciplines seem to publish their own standards. I haven't done any comparing though to check for similarities. There's not an obvious international body.

I'm having trouble with the different Gallons - opinions seem to vary after the 3 or 4 significant digit. I'm starting to think we'd be better off only going to 4 significant digits for these calculations because of that... but have submitted with Quart support & a few tests!

Would it be a bad idea to grab the conversion constants I'm using in multiple places and stick them in functions for reuse? I don't think you've been doing that - so am trying to keep a consistent style.

I should be able to get the Pints and fl oz in today - so this hopefully will be ready to merge.

Happy to alter my non-metric liquid conversion factors to better ones if you manage to find any.